src/app/FakeLib/TargetHelper.fs

```diff
@@ -322,36 +324,114 @@ let listTargets() =
 // Instead of the target can be used the list dependencies graph parameter.
 let doesTargetMeansListTargets target = target = "--listTargets"  || target = "-lt"
 
+/// Determines a parallel build order for the given set of targets
+let determineBuildOrder (d : TargetTemplate<unit>) =
+          
+    // determines the maximal level at which a target occurs
+    let rec determineLevels (currentLevel : int) (levels : Map<string, int>) (t : seq<TargetTemplate<unit>>) =
+        let levels = 
+            t |> Seq.fold (fun m t -> 
+                match Map.tryFind t.Name m with
+                    | Some mapLevel when mapLevel >= currentLevel -> m
+                    | _ -> Map.add t.Name currentLevel m
+            ) levels
+            
+        t |> Seq.map (fun t -> t.Dependencies |> Seq.map getTarget) 
+          |> Seq.fold (determineLevels (currentLevel + 1)) levels
+
+    let levels = determineLevels 0 Map.empty [d]
+
+    // the results are grouped by their level, sorted descending (by level) and 
+    // finally grouped together in a list<TargetTemplate<unit>[]>
+    let result = 
+        levels |> Map.toSeq
+               |> Seq.groupBy snd
+               |> Seq.sortBy (fun (l,_) -> -l)
+               |> Seq.map snd
+               |> Seq.map (fun v -> v |> Seq.map fst |> Seq.distinct |> Seq.map getTarget |> Seq.toArray)
+               |> Seq.toList
+
+    // Note that this build order cannot be considered "optimal" 
+    // since it may introduce order where actually no dependencies
+    // exist. However it yields a "good" execution order in practice.
+    result
+
+/// Runs a single target without its dependencies
+let runSingleTarget (target : TargetTemplate<unit>) =
+    try
+        if errors = [] then
+            traceStartTarget target.Name target.Description (dependencyString target)
+            let watch = new System.Diagnostics.Stopwatch()
+            watch.Start()
+            target.Function()
+            addExecutedTarget target.Name watch.Elapsed
+            traceEndTarget target.Name     
+    with exn ->
+        targetError target.Name exn
+
+/// Runs the given array of targets in parallel using count tasks
+let runTargetsParallel (count : int) (targets : Target[]) =
```

Comment:
Yes this is obviously more concise but I didn't want to use "C# style Linq stuff" here.
I liked your first fragment better for some reason :)